core/cloudflow-operator/src/main/scala/cloudflow/operator/runner/FlinkRunner.scala

```diff
@@ -61,38 +66,28 @@ object FlinkRunner extends Runner[CR] {
 
     import ctx.flinkRunnerSettings._
 
-    val envConfig = EnvConfig(None)
-
     val jobManagerConfig = JobManagerConfig(
       Some(jobManagerSettings.replicas),
-      Resources.make(
-        ResourceRequests.make(jobManagerSettings.resources.memoryRequest.map(_.value),
-                              jobManagerSettings.resources.cpuRequest.map(_.value)),
-        ResourceLimits.make(jobManagerSettings.resources.memoryLimit.map(_.value), jobManagerSettings.resources.cpuLimit.map(_.value))
-      ),
-      Some(envConfig)
+      getJobManagerResourceRequirements(podsConfig, JobManagerPod),
+      Some(EnvConfig(getEnvironmentVariables(podsConfig, JobManagerPod)))
     )
 
     val scale = deployment.replicas
 
     val taskManagerConfig = TaskManagerConfig(
       Some(taskManagerSettings.taskSlots),
-      Resources.make(
-        ResourceRequests.make(taskManagerSettings.resources.memoryRequest.map(_.value),
-                              taskManagerSettings.resources.cpuRequest.map(_.value)),
-        ResourceLimits.make(taskManagerSettings.resources.memoryLimit.map(_.value), taskManagerSettings.resources.cpuLimit.map(_.value))
-      ),
-      Some(envConfig)
+      getTaskManagerResourceRequirements(podsConfig, TaskManagerPod),
+      Some(EnvConfig(getEnvironmentVariables(podsConfig, TaskManagerPod)))
     )
 
     val flinkConfig: Map[String, String] = Map(
-      "state.backend"                  -> "filesystem",
-      "state.backend.fs.checkpointdir" -> s"file://${PVCMountPath}/checkpoints/${deployment.streamletName}",
-      "state.checkpoints.dir"          -> s"file://${PVCMountPath}/externalized-checkpoints/${deployment.streamletName}",
-      "state.savepoints.dir"           -> s"file://${PVCMountPath}/savepoints/${deployment.streamletName}"
-    )
+        "state.backend"                  -> "filesystem",
+        "state.backend.fs.checkpointdir" -> s"file://${PVCMountPath}/checkpoints/${deployment.streamletName}",
+        "state.checkpoints.dir"          -> s"file://${PVCMountPath}/externalized-checkpoints/${deployment.streamletName}",
+        "state.savepoints.dir"           -> s"file://${PVCMountPath}/savepoints/${deployment.streamletName}"
+      ) ++ getFlinkConfig(configSecret)
 
-    val jobSpec = Spec(
+    val _spec = Spec(
```

Comment:
Yes @maasg that is what happens! LOL

Reply:
@RayRoestenburg Could you elaborate on what this is for?

Reply:
Inside joke, @maasg uses `_` sometimes. The change is because this spec is not a jobSpec, it is the spec for the CR, that contains the job(manager)Spec.

Reply:
Oh cool!